File: CTask/src/org/ShapeSHifter420/Fabric/AgeSpecificPushFactory.java
package org.ShapeSHifter420.Fabric;

import org.ShapeSHifter420.Pushes.AgeSpecificPush;
import org.ShapeSHifter420.Pushes.IPush;

import java.util.Map;

public class AgeSpecificPushFactory implements IPushFactory{
    @Override
    public IPush create(Map<String, String> conf) {
        String text = conf.getOrDefault("text", "");
        int age = conf.containsKey("age")? Integer.parseInt(conf.get("age")) : Integer.MAX_VALUE;
        long exp = conf.containsKey("expiry_date")? Long.parseLong(conf.get("expiry_date")): -1;
        return new AgeSpecificPush(text,age,exp);
    }
}

File: CKotlin/Factory/AgeSpecificPushFactory.kt
package Factory

import Pushes.AgeSpecificPush
import Pushes.IPush

object AgeSpecificPushFactory : IPushFactory {
    override fun made(param: Map<String, String>): IPush = AgeSpecificPush(
        age = param["age"]?.toInt() ?: 0,
        expiry = param["expiry_date"]?.toLong() ?: 0,
        text = param["text"]!!
    )
}